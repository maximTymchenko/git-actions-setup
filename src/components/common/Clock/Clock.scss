@import '../../../styles/vars';

.clock {
    cursor: -webkit-grab;
    display: inline-block;
    padding: 5px;
    background-color: $background__secondary;
    font-family: 'OktaNeue-Heavy';
    font-size: 17px;
    font-weight: 900;
    letter-spacing: 0.3px;
    color: $text;
    margin-bottom: 15px;
}
