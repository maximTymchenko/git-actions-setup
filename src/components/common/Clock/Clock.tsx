import React, { FC, useState, useEffect } from 'react';
import moment from 'moment';
import momentTZ from 'moment-timezone';

import './Clock.scss';

interface IClock {
    timezone: any;
}

const Clock: FC = (props: IClock) => {
    const timeZone = momentTZ.tz(`${props.timezone}`).format('hh:mm A');

    const [currentTime, setCurrentTime] = useState(timeZone);

    const updateTime = () => {
        const timeZone = momentTZ.tz(`${props.timezone}`).format('hh:mm A');

        setCurrentTime(timeZone);
    };

    setInterval(updateTime, 1000);

    return (
        <span data-testid="clock" className="clock">
            {currentTime}
        </span>
    );
};

export default Clock;
