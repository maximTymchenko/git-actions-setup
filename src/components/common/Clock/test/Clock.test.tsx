import React from 'react'
import Clock from '../Clock'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'

it('renders elem correctly', () => {
  const { getByTestId } = render(<Clock timezone="Europe/Kiev"></Clock>)
  expect(getByTestId('clock')).toBeTruthy()
})

it('renders without unexpected changes', () => {
  const tree = renderer.create(<Clock timezone="Europe/Kiev"></Clock>).toJSON()
  expect(tree).toMatchSnapshot()
})
