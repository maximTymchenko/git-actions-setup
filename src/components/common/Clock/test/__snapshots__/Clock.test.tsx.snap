// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<span
  className="clock"
  data-testid="clock"
>
  03:13 PM
</span>
`;
